skimage/exposure/_adapthist.py

```diff
@@ -109,40 +116,29 @@ def _clahe(image, ntiles_x, ntiles_y, clip_limit, nbins=128):
     minimum and maximum value as the input image. A clip limit smaller than 1
     results in standard (non-contrast limited) AHE.
     """
-    ntiles_x = min(ntiles_x, MAX_REG_X)
-    ntiles_y = min(ntiles_y, MAX_REG_Y)
 
     if clip_limit == 1.0:
         return image  # is OK, immediately returns original image.
 
-    h_inner = image.shape[0] - image.shape[0] % ntiles_y
-    w_inner = image.shape[1] - image.shape[1] % ntiles_x
+    # make sure we cover the whole image
+    ny = int(np.ceil(image.shape[0] / kernel_size[0]))
+    nx = int(np.ceil(image.shape[1] / kernel_size[1]))
 
-    # make the tile size divisible by 2
-    h_inner -= h_inner % (2 * ntiles_y)
-    w_inner -= w_inner % (2 * ntiles_x)
-
-    orig_shape = image.shape
-    width = w_inner // ntiles_x  # Actual size of contextual regions
-    height = h_inner // ntiles_y
-
-    if h_inner != image.shape[0]:
-        ntiles_y += 1
-    if w_inner != image.shape[1]:
-        ntiles_x += 1
-    if h_inner != image.shape[1] or w_inner != image.shape[0]:
-        h_pad = height * ntiles_y - image.shape[0]
-        w_pad = width * ntiles_x - image.shape[1]
-        image = pad(image, ((0, h_pad), (0, w_pad)), mode='reflect')
-        h_inner, w_inner = image.shape
+    # use same method as ``view_as_windows`` to determine stride size
+    rem = np.array(image.shape) - np.array(kernel_size)
+    height, width = kernel_size
+    while height * (ny - 1) > rem[0]:
+        height -= 1
+    while width * (nx - 1) > rem[1]:
+        width -= 1
```

Comment:
Perhaps add a `best_fit` kwarg option `view_as_windows` to allow it to choose a stepsize that gives full coverage with minimal overlap?